libmscore/layout.cpp

```diff
@@ -1614,8 +1614,32 @@ static void distributeStaves(Page* page)
                         if (!sysStaff->show())
                               continue;
 
+                        Spacer* activeSpacer { nextSpacer };
```

Comment:
@njvdberg can you please exract this new logic into a new method? layout.cpp is already overcomplicated

Reply:
I'll make it in 3.x and will create a dedicated PR for master myself (I'm used to it now :smile: )

Reply:
Done.
See [PR #7722](https://github.com/musescore/MuseScore/pull/7722) for `master`.

For now, only `distrubuteStaves()` will use the new methods but I will replace other places for finding the active spacers by these new methods. However, this will be two separate PR's (for `3.x` and `master`).